src/services/scope-manager.js

```diff
@@ -0,0 +1,76 @@
+import context from '../context';
+
+// 5 minutes exipration cache
+const SCOPE_CACHE_EXPIRATION_DELTA = 300;
+
+class ScopeManager {
+  constructor({ configStore, forestServerRequester, moment } = context.inject()) {
+    this.configStore = configStore;
+    this.forestServerRequester = forestServerRequester;
+    this.moment = moment;
+    this.scopesCache = {};
+  }
+
+  async getScopeForUser(user, collectionName) {
+    if (!user.renderingId) throw new Error('Missing required renderingId');
+    if (!collectionName) throw new Error('Missing required collectionName');
+
+    const collectionScope = await this._getScopeCollectionScope(user.renderingId, collectionName);
+
+    return ScopeManager._formatDynamicValues(user.id, collectionScope);
+  }
+
+  static _formatDynamicValues(userId, collectionScope) {
+    if (!collectionScope?.scope?.filter) return null;
+
+    collectionScope.scope.filter.conditions.forEach((condition) => {
+      if (condition.value
+        && `${condition.value}`.startsWith('$')
+        && collectionScope.scope.dynamicScopesValues.users[userId]) {
+        condition.value = collectionScope
```

Comment:
Are you modifying the entry from the cache without cloning it?
That won't work if multiple users call the service